src/characters/sub/Sub.gd

```diff
@@ -107,15 +105,9 @@ func TriggerLights():
 	yield(get_tree().create_timer(1.0), "timeout")
 
 func DestroySub():
-	if game.debug:
-		print("Sub Is Destroyed")
 	emit_signal("sub_destroyed")
-	
 	input_disabled=true
-	
-	yield(get_tree().create_timer(3),"timeout")
-	game.main.load_screen("res://levels/procedural_test/levelgen.tscn")
```

Comment:
@rainerweston The Sub should only handle the concerns of itself. I don't understand why you went through the trouble of a re-architecting to a scene controller, if you aren't going to delegate responsibility to it. :thinking: